source/extensions/filters/http/ext_proc/mutation_utils.cc

```diff
@@ -40,6 +40,11 @@ absl::Status MutationUtils::applyHeaderMutations(const HeaderMutation& mutation,
                                                  const Checker& checker,
                                                  Counter& rejected_mutations) {
   for (const auto& hdr : mutation.remove_headers()) {
+    if (!Envoy::Http::validHeaderString(hdr)) {
+      ENVOY_LOG(debug, "remove_headers contain null character, may not be removed.");
```

Comment:
Headers can be valid for reasons other than null characters, @yanavlasov

Reply:
You can use the `HeaderUtility::headerNameIsValid` and `HeaderUtility::headerValueIsValid` for RFC compliant check.

Reply:
done